pkg/keylock/keylock.go

```diff
@@ -31,6 +36,10 @@ type KeyLock interface {
 	// Creates the lock if one doesn't already exist.
 	Acquire(id string) LockHandle
 
+	// Acquire a lock associated with the specified ID.
+	// Creates the lock if one doesn't already exist within a particular time
```

Comment:
done

Reply:
I would suggest returning an error here. Since Acquire was guaranteed to return a non-nil LockHandle, we are changing the behavior now with AcquireWithTimeout.  It is much more explicit that the caller needs to handle errors returned back by this function.